### ISSUE TITLE
Specifying an instance by tag

### ISSUE THREAD

**Issue by aidansteele** [glassechidna/ec2connect#3]
Right now we just support `ssh user@<instance-id>`. Perhaps people would want to use `ssh user@<instance name>` or other such tags. How should we do this? A few questions to ponder:

* Should we support only the `Name` tag? 
* Should we support other tags?
* Should we support only one tag or could it be `ssh user@tag=val`?
* Is the above horrific?!
* How should we distinguish between EC2 instances and hostnames on the local network?
* What about tag values with spaces in them?

This is also relevant to #2 as perhaps the jumphost instance is e.g. named `sshjump` in each region.

**Comment by jdub** [comment]
Good call. Building in clever stuff like tag-based lookups would be sweet. (We have an internal tool that does this.)

I have public host names for my instances that I'd prefer to use instead of the instance ID, for which a host name glob would work. You could support VPC host names by default, e.g. `*.compute.internal`.
